### PR TITLE
Update storage metrics, logging and CORS commands

### PR DESCRIPTION
**Pull request by troydai**
1. Enable them property accept account credential from environment.
2. Fix CORS command when services options are not given properly

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by troydai** [comment]
Partially addressed issue https://github.com/Azure/azure-cli/issues/3362

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_factory.py [hunk 1]
```diff
@@ -108,3 +105,30 @@ def cloud_storage_account_service_factory(kwargs):
     sas_token = kwargs.pop('sas_token', None)
     kwargs.pop('connection_string', None)  # pylint: disable=unused-variable
     return CloudStorageAccount(account_name, account_key, sas_token)
+
+
+def multi_service_properties_factory(kwargs):
+    """Create multiple date services properties instance based on the services option"""
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_factory.py]: typo: data services?

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -586,29 +587,77 @@ def register_blob_source_uri_arguments(scope):
...
-
-register_cli_argument('storage cors clear', 'services', help='The storage service(s) for which to clear CORS rules: (b)lob (f)ile (q)ueue (t)able. Can be combined.')
-
-register_cli_argument('storage cors list', 'services', help='The storage service(s) for which to list the CORS rules: (b)lob (f)ile (q)ueue (t)able. Can be combined.')
+#######################################
+# storage cors commands parameters
+#######################################
+
+with CommandContext('storage cors') as c:
+    c.reg_arg('max_age', type=int, default='0',
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: Should default be in the command signature instead of here?

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py [hunk 3]
```diff
@@ -586,29 +587,77 @@ def register_blob_source_uri_arguments(scope):
...
+              help='The number of seconds the client/browser should cache a preflight response.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: Should we move this help text into the _help.py file instead of here, especially given how specialized the CORS commands are.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py [hunk 4]
```diff
@@ -586,29 +587,77 @@ def register_blob_source_uri_arguments(scope):
...
+              help='List of response headers allowed to be part of the cross-origin request.')
+    c.reg_arg('exposed_headers', nargs='+',
+              help='List of response headers to expose to CORS clients.')
+    c.reg_arg('services', validator=get_char_options_validator('bfqt', 'services'),
+              help='The storage service(s) for which to add the CORS rule: (b)lob (f)ile (q)ueue (t)able. '
+                   'Can be combined.')
+
+
+with CommandContext('storage cors list') as c:
+    c.reg_arg('services', validator=get_char_options_validator('bfqt', 'services'), default='bqft', required=False,
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: Can default and requiredness be taken from the command signature?

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py [hunk 5]
```diff
@@ -806,4 +798,20 @@ def services_type(string):
         raise ValueError
     return Services(_str=''.join(set(string)))
 
+
 # endregion
+
+
+def get_char_options_validator(types, property_name):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py]: Can you add some unit tests to cover this validator?

File: src/command_modules/azure-cli-storage/tests/test_storage_account_scenarios.py [hunk 6]
```diff
@@ -80,23 +79,16 @@ def test_logging_operations(self, resource_group, storage_account):
...
-
-        self.cmd('storage metrics update --services f --hour false --retention 1 '
-                 '--connection-string "{}"'.format(connection_string))
+        self.storage_cmd('storage metrics update --services f --hour false --retention 1 ', storage_account_info)
 
-        self.cmd('storage metrics show --connection-string "{}"'.format(connection_string), checks=[
-            JMESPathCheck('file.hour.enabled', False),
-            JMESPathCheck('file.minute.enabled', False),
-        ])
+        self.storage_cmd('storage metrics show', storage_account_info) \
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/tests/test_storage_account_scenarios.py]: Should this test be expanded to test the logic of setting different retention periods for hourly and by-minute metrics?  Or is that not part of this PR?

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_factory.py (hunk 1 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_factory.py]: yes! yes!

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py (hunk 3 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: yes.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py (hunk 5 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py]: ok

File: src/command_modules/azure-cli-storage/tests/test_storage_account_scenarios.py (hunk 6 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/tests/test_storage_account_scenarios.py]: it's not part of this PR.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py (hunk 4 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: For this one, no. The method accept the services as type of `list<SerivceProperties>' instead of `str`

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py (hunk 2 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py]: yes.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py (hunk 5 above)
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py]: Test added.

**Comment by troydai** [comment]
@tjprescott
